### PR TITLE
3rd party provider (baby steps).

### PR DESCRIPTION
**Pull request by Andrewangeta**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -991,9 +991,24 @@ throw Abort(.badRequest)
 throw Abort(.badRequest, reason: “Could not get data from external API.”)
 ```
 
-# 3rd Party Providers
+# 3rd Party Providers 
+When building third party providers for Vapor, it's important to have a certain consistency that users will be able to become famaliar with when switching or adding new providers. Although Vapor is very young there are already certain patterns that make sense when writing providers.
```
Comment (jdmcd) [inline README.md]: Comma after "is very young" please

File: README.md [hunk 2]
```diff
@@ -991,9 +991,24 @@ throw Abort(.badRequest)
...
+### Naming
```
Comment (jdmcd) [inline README.md]: Extra space about this section please

File: README.md [hunk 3]
```diff
@@ -991,9 +991,24 @@ throw Abort(.badRequest)
...
+When naming a provider it's best to name the project itself that will be on github as part of the vapor community organization hyphenated with the extension `-provider`. For example if our provider is named `FooBar` then the project name would be named in the following way:
+`foo-bar-provider`.
+
+When creating a provider library, the Vapor community has agreed to name things while omiting phrases like `Provder` or `Package`. Take the StripeProvider for example, while the name of the project itself can be named `StripeProvider` the library name should be just the product itself:
```
Comment (jdmcd) [inline README.md]: Could we remove "the Vapor community has agreed?" That sentence sounds weird to me. Maybe something like "When creating a provider library, you should omit phrases like..."

**Review by jdmcd — CHANGES_REQUESTED** [review]
Overall this looks great! Just a few small changes here and there and then we can merge!

File: README.md (hunk 3 above)
Comment (Andrewangeta) [reply README.md]: Yea i was questioning that myself lol

**Comment by Andrewangeta** [comment]
@mcdappdev Should be good to go 👍

**Review by jdmcd — APPROVED** [review]

**Comment by jdmcd** [comment]
@Andrewangeta You're the best! Thanks for taking care of that section.

**Pull request merged**